src/core/vanilla.ts

```diff
@@ -453,7 +453,25 @@ const writeAtomState = <Value, Update>(
             // NOTE technically possible but restricted as it may cause bugs
             throw new Error('no atom init')
           }
-          setAtomValue(state, a, v)
+          if (v instanceof Promise) {
+            const promise = v
+              .then((vv) => {
```

Comment:
Maybe some better naming than v, vv and a.

Reply:
Thanks. `a` and `v` are used for other places, so let's tackle it separately. (I agree to fix because it doesn't harm to bundle size. suggestions are welcome.)
For `vv`, let me fix in this PR.